### PR TITLE
[boost-di] Update to 1.2.0

### LINKED ISSUE #15280: [boost-di] update to 1.2
Library name: boost-di

New version number: 1.2

Please update the port. It's more than a year old.

### PR DESCRIPTION
**Pull request by PhoebeHui**
Fixes https://github.com/microsoft/vcpkg/issues/15280

Update boost-di to latest release revision 1.2.0.

### CODE REVIEW

**Review by NancyLi1013 — APPROVED** [review]

**Comment by StarGate-One** [comment]
@PhoebeHui @NancyLi1013 
- You may want to change references to `boost-experimental` to `boost-ext`
   - I looks as if the library has moved to https://github.com/boost-ext/di
   - https://github.com/boost-experimental/di gives me a redirect back over to https://github.com/boost-ext/di

File: ports/boost-di/portfile.cmake [hunk 1]
```diff
@@ -1,17 +1,17 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO boost-experimental/di
```
Comment (StarGate-One) [inline ports/boost-di/portfile.cmake:3]: I would change this to `boost-ext/di` as it looks like the maintainers have moved everything for `boost-experimental` to `boost-ext` on github

Comment (NancyLi1013) [reply ports/boost-di/portfile.cmake:3]: @StarGate-One Thanks for pointing this out. Sorry, we missed this. As you said above, `boost-experimental/di` doesn't exist now. We should update it to the new repo.

Comment (NancyLi1013) [inline ports/boost-di/portfile.cmake:3]:
```suggestion
    REPO boost-ext/di
```

File: ports/boost-di/CONTROL [hunk 2]
```diff
@@ -1,4 +1,4 @@
 Source: boost-di
-Version: 1.1.0-1
+Version: 1.2.0
 Homepage: https://github.com/boost-experimental/di
```
Comment (NancyLi1013) [inline ports/boost-di/CONTROL:3]:
```suggestion
Homepage: https://github.com/boost-ext/di
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/boost-di/portfile.cmake (hunk 1 above)
Comment (StarGate-One) [reply ports/boost-di/portfile.cmake:3]: You are most welcome!

**Review by NancyLi1013 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thank you :D

**Pull request merged**